### PR TITLE
feat: fetch squads via graphql instead of from boot

### PR DESCRIPTION
**Pull request by kopancek**
## Changes

- added new useSquads hook
- removed all squad references from boot and AuthContext
- using the new useSquads hook everywhere where we used AuthContext squads before
- no visual changes

## Important

Without the API PR being merged and deployed, it will not work in preview or extension:
- https://github.com/dailydotdev/daily-api/pull/1879

## Events

No new events

### On those affected packages:
- [X] Have you done sanity checks in the webapp?
- [x] Have you done sanity checks in the extension?
- [x] Does this not break anything in companion?

### Did you test the modified components media queries?
- [X] MobileL (420px)
- [X] Tablet (656px)
- [X] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-324 #done

### Preview domain
https://mi-324-do-not-use-squads-in-boot.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/graphql/squads.ts
Comment (rebelchris) [inline packages/shared/src/graphql/squads.ts:215]: I think there's also a option to pass it as infinite maybe more failsafe (although don't think anyone but daphna will ever hit 100+ 🤣 )

File: packages/shared/src/hooks/squads/useSquads.ts
Comment (rebelchris) [inline packages/shared/src/hooks/squads/useSquads.ts:47]: Why did you disable retry specifically?
We also have a wrapper for `disabledRefetch` which maybe you want?

File: packages/shared/src/hooks/useDeleteSquad.ts
Comment (rebelchris) [inline packages/shared/src/hooks/useDeleteSquad.ts:22]: This whole part feels it could be part of useSquad, how did you feel about it looking at code?
(Don't need to refactor, just looking for your opinion as someone who touched the code)

**Review by rebelchris — APPROVED** [review]
Nice work!

Can we maybe do a bit longer internal test on this one?
I think we have to be careful with mobile sharing from various places/extension/companion adoption etc.

It's easy to have forgotten a place where squads where used potentially.
Code wise nothing much to add though 🙏

**Review by nensidosari — CHANGES_REQUESTED** [review]
Looks great! Nothing to add 🚀 
Just blocking the PR until the api PR is merged so we can test before merging or to avoid any accidental merge.

File: packages/shared/src/graphql/squads.ts
Comment (kopancek) [reply packages/shared/src/graphql/squads.ts:215]: AFAIK, we limit to 100 by default on the API side anyway: https://github.com/dailydotdev/daily-api/blob/main/src/common/datePageGenerator.ts#L39
https://github.com/dailydotdev/daily-api/blob/main/src/schema/sources.ts#L878-L883

IMO, we cannot even show 100 items in the sidebar anyway, so having this paginated probably makes sense.

File: packages/shared/src/hooks/squads/useSquads.ts
Comment (kopancek) [reply packages/shared/src/hooks/squads/useSquads.ts:47]: Bad copy paste from the `useSquad` hook. I think we can retry here.

**Comment by kopancek** [comment]
> Can we maybe do a bit longer internal test on this one? I think we have to be careful with mobile sharing from various places/extension/companion adoption etc.
> 
> It's easy to have forgotten a place where squads where used potentially. Code wise nothing much to add though 🙏

Yup, for sure. I tried to test in extension/companion locally, but since it does not use local API endpoint I could only see that the requests are firing. (Hence why I kept the checkboxes on the PR descriptions unchecked).

Still waiting on the [CI to deploy the API changes](https://app.circleci.com/pipelines/github/dailydotdev/daily-api?branch=main).

File: packages/shared/src/hooks/useDeleteSquad.ts
Comment (kopancek) [reply packages/shared/src/hooks/useDeleteSquad.ts:22]: For me, the `useDeleteSquad` hook is more about showing a prompt confirmation when deleting a squad. So maybe we can rename it to `useDeleteSquadPrompt` instead. That way it should be clearer.

I'd rather refrain from refactoring the hook in this PR, since it's already touching a lot of code. But if you feel like we should have the functionality in useSquad hook, let's add a tech debt issue for that :)

**Review by nensidosari — APPROVED** [review]

**Pull request merged**
